test/test_pallas.py

```diff
@@ -920,6 +922,106 @@ def test_flash_attention_sm_scale_backward(self):
       self.assertTrue(torch.allclose(i[0].grad.cpu(), i[1].cpu(), atol=1e-05))
     jax.config.update("jax_default_matmul_precision", "default")
 
+  @unittest.skipIf(xr.device_type() != 'TPU' or tpu.version() < 3,
+                   "This test only works on TPUv3+.")
+  def test_flash_attention_ab(self):
+    jax.config.update("jax_default_matmul_precision", "highest")
```

Comment:
lol we should really make a context manager that take cares of this in this test.